src/toil/test/src/jobCacheTest.py

```diff
@@ -252,9 +252,9 @@ def _testCacheEviction(self, file1MB, file2MB, diskRequestMB):
                 Job.Runner.startToil(A, self.options)
             except FailedJobsException as err:
                 self.assertEqual(err.numberOfFailedJobs, 1)
-                errType, errMsg = self._parseAssertionError(self.options.logFile)
-                if (errType == 'AssertionError' and
-                        errMsg == 'Unable to free up enough space for caching.'):
+                with open(self.options.logFile) as f:
+                    logContents = f.read()
+                if 'AssertionError: Unable to free up enough space for caching.' in logContents:
```

Comment:
This is a step in the right direction IMO, but you're right it's still brittle. The current approach's weak point is 
https://github.com/BD2KGenomics/toil/blob/master/src/toil/test/src/jobCacheTest.py#L353
where is tests for a particular log message from leader.py, and this is mainly what I wanted to remove. The parser, when successful, returned the exception message which was then compared against another hardcoded string. I'm still relying on a hardcoded string but the test now would only break if you touch caching code, specifically that particular assertion.

I could also define a caching specific exception to raise and test for it in the output, that would probably be more flexible. Ideally though my approach wouldn't rely on output parsing at all, I'll keep thinking about it.